pkg/authn/jwks/authn.go

```diff
@@ -105,45 +129,63 @@ func (a authentication) Authenticate(ctx context.Context, token string) (authn.S
 	}, nil
 }
 
-func (a authentication) fetchJWKS(ctx context.Context) (jwk.Set, error) {
-	jwksCache.RLock()
-	if time.Since(jwksCache.cachedAt) < cacheDuration && jwksCache.jwks.Len() > 0 {
-		cached := jwksCache.jwks
-		jwksCache.RUnlock()
-		return cached, nil
+func isSignatureError(err error) bool {
+	return !errors.Contains(err, errJWTExpiryKey) &&
+		!errors.Contains(err, errInvalidIssuer) &&
+		!errors.Contains(err, smqauth.ErrExpiry)
+}
+
+func (a authentication) fetchJWKS(ctx context.Context, forceRefresh bool) (jwk.Set, error) {
```

Comment:
I guess some library will be having this function 
But if we going to like `NATS team`, then we can have this functions